pkg/sql/schemachanger/scplan/plan.go

```diff
@@ -128,22 +129,48 @@ func makePlan(ctx context.Context, p *Plan) (err error) {
 	return nil
 }
 
+func applyOpRules(
+	ctx context.Context, activeVersion clusterversion.ClusterVersion, g *scgraph.Graph,
+) (*scgraph.Graph, error) {
+	if activeVersion.IsActive(clusterversion.V23_1) {
+		return current.ApplyOpRules(ctx, g)
+	} else if activeVersion.IsActive(clusterversion.V22_2) {
+		return release_22_2.ApplyOpRules(ctx, g)
+	} else {
+		log.Warningf(ctx, "Falling back to the oldest supported version 22.2")
+		return release_22_2.ApplyOpRules(ctx, g)
+	}
+}
+
+func applyDepRules(
+	ctx context.Context, activeVersion clusterversion.ClusterVersion, g *scgraph.Graph,
+) error {
+	if activeVersion.IsActive(clusterversion.V23_1) {
+		return current.ApplyDepRules(ctx, g)
+	} else if activeVersion.IsActive(clusterversion.V22_2) {
+		return release_22_2.ApplyDepRules(ctx, g)
+	} else {
+		log.Warningf(ctx, "Falling back to the oldest supported version 22.2")
+		return release_22_2.ApplyDepRules(ctx, g)
+	}
+}
```

Comment:
nit: make a function to look up the registry for the version rather than dispatching each of these methods.

Reply:
Good point, added a new function and use that here:

// getRulesRegistryForRelease returns the rules registry based on the current
// active version of cockroach. In a mixed version state, it's possible for the state
// generated by a newer version of cockroach to be incompatible with old versions.
// For example dependent objects or combinations of them in a partially executed,
// plan may reach states where older versions of cockroach may not be able
// to plan further (and vice versa). To eliminate the possibility of these issues, we
// will plan with the set of rules belonging to the currently active version. One
// example of this is the dependency between index name and secondary indexes
// is more relaxed on 23.1 vs 22.2, which can lead to scenarios where the index
// name may become public before the index is public (which was disallowed on older
// versions).